MapboxNavigation/RouteTableViewController.swift

```diff
@@ -68,20 +68,41 @@ class RouteTableViewController: UIViewController {
 extension RouteTableViewController: UITableViewDelegate, UITableViewDataSource {
 
     func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+        return routeController.routeProgress.route.legs.count
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        // Don't display section header if there is only one step
+        guard routeController.routeProgress.route.legs.count > 1 else {
+            return nil
+        }
+        let leg = routeController.routeProgress.route.legs[section]
+        
+        let sourceName = leg.source.name
+        let destinationName = leg.destination.name
+        let legToFrom = leg.name.components(separatedBy: ", ")
+        
+        if let destinationName = destinationName?.nonEmptyString, legToFrom.count > 1 {
+            return "\(destinationName) via \(legToFrom.joined(separator: " and "))"
```

Comment:
It would be easier to check for `leg.name` containing `, ` and replace `, ` with ` and `. But in any case, we should use a localizable format string instead of string interpolation.